library/ziti_enroll.c

```diff
@@ -181,31 +179,30 @@ static void well_known_certs_cb(char *base64_encoded_pkcs7, ziti_error *err, voi
     if (strcmp("ott", enroll_req->ecfg->zej->method) == 0) {
         size_t len;
         if (enroll_req->ecfg->private_key == NULL) {
-            tls->api->generate_key(&enroll_req->ecfg->pk);
-            tls->api->write_key_to_pem(enroll_req->ecfg->pk, &enroll_req->ecfg->private_key, &len);
+            ziti_err = ZITI_KEY_GENERATION_FAILED;
+            TRY(TLS, tls->api->generate_key(&enroll_req->ecfg->pk));
+            TRY(TLS, tls->api->write_key_to_pem(enroll_req->ecfg->pk, &enroll_req->ecfg->private_key, &len));
         }
         else {
-            tls->api->load_key(&enroll_req->ecfg->pk, enroll_req->ecfg->private_key,
-                               strlen(enroll_req->ecfg->private_key));
+            ziti_err = ZITI_KEY_LOAD_FAILED;
+            TRY(TLS, tls->api->load_key(&enroll_req->ecfg->pk, enroll_req->ecfg->private_key,
+                                        strlen(enroll_req->ecfg->private_key)));
```

Comment:
strlen + 1 not needed? (the change you made to unblock me yesterday)

Reply:
yep, I was testing errors

Reply:
fixed